### PR TITLE
Fix: update tsconfig and log message

### LINKED ISSUE #18984: PiT 24.4: when migrating a project to 24.4 modifications to tsconfig.json and types.d.ts should throw once
### Description of the bug

When frontend configuration files require updating, Flow does not perform these updates in a batch. Instead, it executes separate modifications and throws errors, necessitating the user to run the application multiple times after the migration.

NOTE: it makes several projects run with the PiT script fail because it only considers one extra run

### Expected behavior

It should throw just once

### Minimal reproducible example

The attached project is a 24.3 project with the needed modifications to run in 24.4, but types.d.ts, tsconfig.json and package.json have not been modified.
Unzip the project and run mvn, you need to run 3 times

[latest-lit-top.zip](https://github.com/vaadin/flow/files/14647570/latest-lit-top.zip)

### Versions

- Vaadin / Flow version: 24.4.0.alpha15

### PR DESCRIPTION
**Pull request by caalador**
Instead of updating and throwing
exception for 'tsconfig.json'.
Generate .bak file,
update file and log info if the
file is updated.

Fixes #18984

### CODE REVIEW

**Comment by mshabarov** [comment]
I'd suggest to also align/unify how back-up file is created and which format is used for `tsconfig.json` and `types.d.ts`.
Also, it would make sense to log/throw only once in the end, giving a bunch of messages to the user.

See my comments here https://github.com/vaadin/flow/issues/18984#issuecomment-2006594242.

**Comment by mshabarov** [comment]
Collecting the logs can be done separately, as now we have the [issue](https://github.com/vaadin/flow/issues/18986) for it :)

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsConfig.java [hunk 1]
```diff
@@ -52,15 +58,17 @@ public class TaskGenerateTsConfig extends AbstractTaskClientGenerator {
...
-    static final String ERROR_MESSAGE =
-            "%n%n**************************************************************************"
-            + "%n*  TypeScript config file 'tsconfig.json' has been updated to the latest *"
-            + "%n*  version by Vaadin. Please verify that the updated 'tsconfig.json'     *"
-            + "%n*  file contains configuration needed for your project (add missing part *"
-            + "%n*  from the old file if necessary) and restart the application.          *"
-            + "%n**************************************************************************%n%n";
-    //@formatter:on
+    static final String ERROR_MESSAGE = """
+
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsConfig.java:62]: nit: This has one line break before and after, whereas the similar banner for types.d.ts has two line breaks, would it be better to align them so they both have one or two lines each?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java [hunk 2]
```diff
@@ -171,8 +176,10 @@ private void updateIfContentMissing() throws ExecutionFailedException {
                         Files.writeString(tsDefinitions.toPath(),
                                 uncommentedDefaultContent,
                                 StandardOpenOption.APPEND);
-                        if (updateMode == UpdateMode.UPDATE_AND_THROW) {
-                            throw new ExecutionFailedException(UPDATE_MESSAGE);
+                        if (updateMode == UpdateMode.UPDATE_AND_BACKUP
+                                && !warningEmitted) {
+                            log().warn(UPDATE_MESSAGE);
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java:181]: Misses String.format: `log().warn(String.format(UPDATE_MESSAGE));`
Or `TaskGenerateTsConfig` has redundant String.format, hmm

**Review by mshabarov — DISMISSED** [review]
Works as expected to me: messages are logged one time, each has backup file, each has the same format of the file and unique ID.

Couple of nits, feel free to merge, if you think they are non-sense.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsConfig.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsConfig.java:62]: Aligned both to have the same single as used normally.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java (hunk 2 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java:181]: Removed unneeded string format

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java [hunk 3]
```diff
@@ -43,7 +42,6 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
     private static final String DECLARE_CSS_MODULE = "declare module '*.css?inline' {";
     static final String UPDATE_MESSAGE = """
 
-
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java:46]: This message should contains old "flowBackup" suffix. Should contain the same text, as for tsconfig: "Old configuration is stored as a '.bak' file."

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java (hunk 3 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java:46]: done

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java [hunk 4]
```diff
@@ -43,21 +42,18 @@ public class TaskGenerateTsDefinitions extends AbstractTaskClientGenerator {
...
             ***************************************************************************
             *  The TypeScript type declaration file 'types.d.ts' has been updated     *
             *  to the latest version by Vaadin. Previous content has been backed up   *
-            *  on 'types.d.ts.flowBackup' file. Please verify that the updated        *
-            *  'types.d.ts' file contains configuration needed for your project, and  *
-            *  then delete the backup file.                                           *
+            *  as a '.bak' file.. Please verify that the updated 'types.d.ts' file    *
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateTsDefinitions.java:48]:
```suggestion
            *  as a '.bak' file. Please verify that the updated 'types.d.ts' file    *
```

**Review by mshabarov — APPROVED** [review]

**Pull request merged**